src/event/entity/EntityTeleportEvent.php

```diff
@@ -35,10 +35,18 @@
 class EntityTeleportEvent extends EntityEvent implements Cancellable{
 	use CancellableTrait;
 
+	public const CAUSE_PLUGIN = 0;
+	public const CAUSE_ITEM = 1;
+	public const CAUSE_WORLD = 2;
+	public const CAUSE_RESPAWN = 3;
+	public const CAUSE_PATCH = 4;
```

Comment:
This naming is very unintuitive, I have no idea what it's supposed to mean

Reply:
- Item: this is when an item cause the teleportation, like chorus fruit or enderpearl
- World: this is when the world cause teleportation like changing world or being teleported in safe spawn due to world unload
- Respawn: this is when the player respawn cause teleportation
- Patch: I think we can merge it with "Respawn" as "Player"
So, I can suggest this changes:
```diff
	public const CAUSE_ITEM = 1;
	public const CAUSE_WORLD = 2;
-	public const CAUSE_RESPAWN = 3;
-	public const CAUSE_PATCH = 4;
+	public const CAUSE_PLAYER = 3;
```

Reply:
This is even less clear than the initial, and additionally provides less information.